Uninformative crash when project is run without j-ecdar files

When users clone the repository and run it without adding additional j-ecdar files to `/lib/`, they are met with a null-pointer exception because the gui tries to show the error in a snackbar on a presentation that is uninitialized. 
We should probably catch the error somewhere else and display it to the user, either in the GUI or the console, as it may be a common error.

The exception occurs here in `Ecdar.java` with a null `presentation`: 
```java
public static void showToast(final String message) {
        presentation.showSnackbarMessage(message);
}
```

The stacktrace is here and is very uninformative of the actual error:
```
Exception in Application start method
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
	at ecdar.presentations.FilePresentation.lambda$new$0(FilePresentation.java:45)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	at java.base/java.lang.Thread.run(Thread.java:829)
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:465)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:364)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:1051)
Caused by: java.lang.RuntimeException: Exception in Application start method
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:901)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.RuntimeException: Exception in Application start method

Caused by: java.lang.IllegalStateException: javafx.fxml.LoadException: 
/C:/Users/sebas/Desktop/Linux/Ecdar-GUI/build/resources/main/ecdar/presentations/EcdarPresentation.fxml:542

	at ecdar.presentations.EcdarFXMLLoader.loadAndGetController(EcdarFXMLLoader.java:30)
Caused by: java.lang.IllegalStateException: javafx.fxml.LoadException: 

	at ecdar.presentations.EcdarPresentation.<init>(EcdarPresentation.java:59)
	at ecdar.Ecdar.start(Ecdar.java:208)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:847)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:484)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	... 1 more
Caused by: javafx.fxml.LoadException: 
/C:/Users/sebas/Desktop/Linux/Ecdar-GUI/build/resources/main/ecdar/presentations/EcdarPresentation.fxml:542

	at javafx.fxml/javafx.fxml.FXMLLoader.constructLoadException(FXMLLoader.java:2707)
Caused by: javafx.fxml.LoadException: 

	at javafx.fxml/javafx.fxml.FXMLLoader$InstanceDeclarationElement.constructValue(FXMLLoader.java:1022)
	at javafx.fxml/javafx.fxml.FXMLLoader$ValueElement.processStartElement(FXMLLoader.java:757)
	at javafx.fxml/javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2808)
	at javafx.fxml/javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2634)
	at javafx.fxml/javafx.fxml.FXMLLoader.load(FXMLLoader.java:2532)
	at ecdar.presentations.EcdarFXMLLoader.loadAndGetController(EcdarFXMLLoader.java:28)
	... 11 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at javafx.fxml/javafx.fxml.FXMLLoader$InstanceDeclarationElement.constructValue(FXMLLoader.java:1020)
Caused by: java.lang.reflect.InvocationTargetException

	... 16 more
Caused by: java.lang.IllegalStateException: javafx.fxml.LoadException: 
/C:/Users/sebas/Desktop/Linux/Ecdar-GUI/build/resources/main/ecdar/presentations/BackendOptionsDialogPresentation.fxml

Caused by: java.lang.IllegalStateException: javafx.fxml.LoadException: 

	at ecdar.presentations.EcdarFXMLLoader.loadAndGetController(EcdarFXMLLoader.java:30)
	at ecdar.presentations.BackendOptionsDialogPresentation.<init>(BackendOptionsDialogPresentation.java:10)
	... 21 more
Caused by: javafx.fxml.LoadException: 
/C:/Users/sebas/Desktop/Linux/Ecdar-GUI/build/resources/main/ecdar/presentations/BackendOptionsDialogPresentation.fxml

Caused by: javafx.fxml.LoadException: 

	at javafx.fxml/javafx.fxml.FXMLLoader.constructLoadException(FXMLLoader.java:2707)
	at javafx.fxml/javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2685)
	at javafx.fxml/javafx.fxml.FXMLLoader.load(FXMLLoader.java:2532)
	at ecdar.presentations.EcdarFXMLLoader.loadAndGetController(EcdarFXMLLoader.java:28)
	... 22 more
Caused by: java.lang.NullPointerException
	at ecdar.Ecdar.showToast(Ecdar.java:127)
Caused by: java.lang.NullPointerException

	at ecdar.controllers.BackendOptionsDialogController.setBackendPathIfFileExists(BackendOptionsDialogController.java:212)
	at ecdar.controllers.BackendOptionsDialogController.getPackagedBackends(BackendOptionsDialogController.java:191)
	at ecdar.controllers.BackendOptionsDialogController.initializeBackendInstanceList(BackendOptionsDialogController.java:106)
	at ecdar.controllers.BackendOptionsDialogController.initialize(BackendOptionsDialogController.java:43)
	at javafx.fxml/javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2655)
	... 24 more
Exception running application ECDAR

> Task :run FAILED

Execution failed for task ':run'.
> Process 'command 'C:\Program Files\Zulu\zulu-11\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

--- a/src/main/java/ecdar/Ecdar.java
+++ b/src/main/java/ecdar/Ecdar.java
@@ -124,7 +124,9 @@ public class Ecdar extends Application {
     }
 
     public static void showToast(final String message) {
-        presentation.showSnackbarMessage(message);
+        Platform.runLater(() -> {
+            presentation.showSnackbarMessage(message);
+        });
     }
 
     public static void showHelp() {

--- a/src/main/java/ecdar/controllers/BackendOptionsDialogController.java
+++ b/src/main/java/ecdar/controllers/BackendOptionsDialogController.java
@@ -208,14 +208,7 @@ public class BackendOptionsDialogController implements Initializable {
             }
         }
 
-        if (engine.getBackendLocation().equals("")) {
-            Ecdar.showToast("File for packaged backend: " + engine.getName() +
-                    " could not be loaded. Please make sure that at least one of the following files is placed in the 'lib' directory: " +
-                    potentialFiles.stream().map(File::getPath).collect(Collectors.joining(", ")));
-            return false;
-        }
-
-        return true;
+        return !engine.getBackendLocation().equals("");
     }
 
     /**

--- a/src/main/java/ecdar/controllers/EcdarController.java
+++ b/src/main/java/ecdar/controllers/EcdarController.java
@@ -242,9 +242,12 @@ public class EcdarController implements Initializable {
             }
         });
 
-        // Set default backend instance
-        BackendInstance defaultBackend = BackendHelper.getBackendInstances().stream().filter(BackendInstance::isDefault).findFirst().orElse(BackendHelper.getBackendInstances().get(0));
-        BackendHelper.setDefaultBackendInstance(defaultBackend);
+        if (BackendHelper.getBackendInstances().size() < 1) {
+            Ecdar.showToast("No engines were found. Download j-Ecdar or Reveaal, or add another engine to fix this. No queries can be executed without engines.");
+        } else {
+            BackendInstance defaultBackend = BackendHelper.getBackendInstances().stream().filter(BackendInstance::isDefault).findFirst().orElse(BackendHelper.getBackendInstances().get(0));
+            BackendHelper.setDefaultBackendInstance(defaultBackend);
+        }
     }
 
     private void initializeDialog(JFXDialog dialog, StackPane dialogContainer) {
@@ -660,10 +663,10 @@ public class EcdarController implements Initializable {
         menuBarViewCanvasSplit.setOnAction(event -> {
             final BooleanProperty isSplit = Ecdar.toggleCanvasSplit();
             if (isSplit.get()) {
-                setCanvasModeToSingular();
+                Platform.runLater(this::setCanvasModeToSingular);
                 menuBarViewCanvasSplit.setText("Split canvas");
             } else {
-                setCanvasModeToSplit();
+                Platform.runLater(this::setCanvasModeToSplit);
                 menuBarViewCanvasSplit.setText("Merge canvases");
             }
         });
@@ -907,7 +910,8 @@ public class EcdarController implements Initializable {
         if (model != null) {
             canvasShellPresentation.getController().canvasPresentation.getController().setActiveModel(activeCanvasPresentation.get().getController().getActiveModel());
         } else {
-            canvasShellPresentation.getController().canvasPresentation.getController().setActiveModel(Ecdar.getProject().getComponents().get(0));
+            // If no components where found, the project has not been initialized. The active model will be updated when the project is initialized
+            canvasShellPresentation.getController().canvasPresentation.getController().setActiveModel(Ecdar.getProject().getComponents().stream().findFirst().orElse(null));
         }
 
         canvasPane.getChildren().add(canvasShellPresentation);
@@ -1177,7 +1181,6 @@ public class EcdarController implements Initializable {
 
     /**
      * This method is used to push the contents of the file and query panes when the tab pane is opened
-     *
      */
     private void changeInsetsOfFileAndQueryPanes() {
         if (messageTabPane.getController().isOpen()) {

--- a/src/main/java/ecdar/presentations/EcdarPresentation.java
+++ b/src/main/java/ecdar/presentations/EcdarPresentation.java
@@ -255,6 +255,9 @@ public class EcdarPresentation extends StackPane {
     }
 
     private void initializeToggleQueryPaneFunctionality() {
+        initializeOpenQueryPaneAnimation();
+        initializeCloseQueryPaneAnimation();
+
         // Set the translation of the query pane to be equal to its width
         // Will hide the element, and force it in then the right side of the border pane is enlarged
         controller.queryPane.translateXProperty().bind(controller.queryPane.widthProperty());
@@ -322,6 +325,9 @@ public class EcdarPresentation extends StackPane {
     }
 
     private void initializeToggleFilePaneFunctionality() {
+        initializeOpenFilePaneAnimation();
+        initializeCloseFilePaneAnimation();
+
         // Set the translation of the file pane to be equal to its width
         // Will hide the element, and force it in then the left side of the border pane is enlarged
         controller.filePane.translateXProperty().bind(controller.filePane.widthProperty().multiply(-1));
@@ -459,7 +465,7 @@ public class EcdarPresentation extends StackPane {
 
     public void showSnackbarMessage(final String message) {
         JFXSnackbarLayout content = new JFXSnackbarLayout(message);
-        controller.snackbar.enqueue(new JFXSnackbar.SnackbarEvent(content, new Duration(3000)));
+        controller.snackbar.enqueue(new JFXSnackbar.SnackbarEvent(content, new Duration(5000)));
     }
 
     public void showHelp() {

